ironfish/src/network/messages/getBlockHeaders.ts

```diff
@@ -24,7 +30,16 @@ export class GetBlockHeadersRequest extends RpcNetworkMessage {
 
   serialize(): Buffer {
     const bw = bufio.write(this.getSize())
-    bw.writeU32(this.start)
+
+    const isBuffer = Buffer.isBuffer(this.start)
+    bw.writeU8(Number(isBuffer))
+
+    if (isBuffer) {
+      bw.writeHash(this.start as Buffer)
+    } else {
+      bw.writeU32(this.start as number)
+    }
```

Comment:
Nitpicky, but would it work to do this? Main reason I'd ask is I prefer to avoid `as` casting where possible since it overrides the type system without any checks.

```suggestion
    if (Buffer.isBuffer(this.start)) {
      bw.writeU8(1)
      bw.writeHash(this.start)
    } else {
      bw.writeU8(0)
      bw.writeU32(this.start)
    }
```